### DISCUSSION TITLE
Stunner v.016 и v.017 and GKE Autopilot 1.27

### DISCUSSION THREAD

**Discussion by dgkolkovskiy** [l7mp/stunner#123] [Q&A]
Hello everyone,

Currently, we have updated GKE Autopilot to version 1.27, and this has broken the stunner. From GKE 1.27, CRDs can't be modified, as they are part of the managed configuration of the Google cluster. 

Any ideas how to fix that?

**Comment by rg0now** [comment]
Could you please post the source of the info "from GKE 1.27, CRDs can't be modified"? Our understanding is that in general you can add any CRD you like, it is that it is "only" the Gateway API CRDs that can't be installed. This is because GKE seems to be stuck at the Gateway API v1beta1 while we have already moved to v1 and you can't have two "default" versions installed at the same time. 

Note that we're not the only project suffering from this, see here: https://github.com/envoyproxy/gateway/issues/2301. The suggested workarounds:
- use GKE standard clusters and disable the Gateway API packaged by Google,
- install STUNner v0.16.0 that was based on Gateway API v1beta1,
- wait for GKE to upgrade to Gateway API v1.

**Reply by dgkolkovskiy** [reply]
Unfortunately, we can't disable the Gateway API packaged by Google, GKE autopilot doesn't allow to do that

**Reply by rg0now** [reply]
Yes, the "use GKE standard clusters and disable the Gateway API packaged by Google" solution only works in [GKE "standard mode"](https://cloud.google.com/kubernetes-engine/docs/concepts/choose-cluster-mode) clusters.
